src/gui/qgsdatadefinedbutton.cpp

```diff
@@ -394,6 +404,21 @@ void QgsDataDefinedButton::showDescriptionDialog()
   mv->exec();
 }
 
+void QgsDataDefinedButton::showAsssistant()
+{
+  if ( mAssistant->exec() == QDialog::Accepted )
+  {
+    const QString newExp = mAssistant->expressionText();
```

Comment:
Why not make QgsDataDefinedAssistant return a QgsDataDefined instead of just a string? That way assistants could be created which fields, deactivated expressions, etc. It would also be more future proof when QgsDataDefined is extended to support additional features (such as time based/animation support).